sql/moz-fx-data-shared-prod/mdn_yari_derived/mdn_popularities_v1/query.py

```diff
@@ -0,0 +1,117 @@
+  host_stripped.page_hits / FIRST_VALUE(host_stripped.page_hits) OVER (
+    ORDER BY
+      page_hits DESC
+  ) AS popularity
+FROM
+  host_stripped
+WHERE
+  ARRAY_LENGTH(SPLIT(host_stripped.`path`, '/')) > 3
+ORDER BY
+    popularity DESC
+"""
+
+APP = "mdn_yari"
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument(
+    "--date", type=lambda x: datetime.strptime(x, "%Y-%m-%d").date(), required=True
+)
+parser.add_argument("--project", default="moz-fx-data-shared-prod")
+parser.add_argument("--temp_dataset", default="tmp")
+parser.add_argument("--temp_bucket", default="moz-fx-data-prod-external-data")
+parser.add_argument("--temp_table", default="mdn_popularities_v1")
+parser.add_argument("--destination_project", default="moz-fx-dev-gsleigh-migration")
+parser.add_argument("--destination_bucket", default="mdn-dev")
+parser.add_argument("--destination_path", default="popularities/current")
+
+
+def main():
+    """Generate a tmp table and extracts to GCS."""
+    args = parser.parse_args()
+    client = bigquery.Client(args.project)
+
+    # create temp table
+    temp_table = f"{args.project}.{args.temp_dataset}.{args.temp_table}"
+    logging.info("Creating a temp table: %s" % temp_table)
+
+    client.query(
+        QUERY_TEMPLATE.format(date=args.date),
+        job_config=bigquery.QueryJobConfig(
+            destination=temp_table,
+            write_disposition=bigquery.job.WriteDisposition.WRITE_TRUNCATE,
+        ),
+    ).result()
+
+    # table to GCS object
+    dataset_ref = bigquery.DatasetReference(args.project, args.temp_dataset)
+    table_ref = dataset_ref.table(args.temp_table)
+
+    target_file_name = f"{args.date.strftime('%m-%Y')}.json"
+    temp_uri = (
+        f"gs://{args.temp_bucket}/{APP}/{args.destination_path}/{target_file_name}.json"
+    )
+
+    extract_job = client.extract_table(
+        table_ref,
+        temp_uri,
+        location="US",
+    )
+    extract_job.result()  # Waits for job to complete.
+    logging.info("Exporting %s to GCS: %s" % (temp_table, temp_uri))
+
+    client.delete_table(temp_table)
+    logging.info("Deleting temp table: %s" % temp_table)
+
+    # download the GCS object
+    storage_client = storage.Client(args.project)
+    bucket = storage_client.bucket(args.temp_bucket)
+    local_path = f"/tmp/{target_file_name}.json"
+    temp_blob_uri = f"{APP}/{args.destination_path}/{target_file_name}.json"
+
+    logging.info("Downloading: %s to %s" % (temp_uri, local_path))
+    temp_blob = bucket.blob(temp_blob_uri)
+    temp_blob.download_to_filename(local_path)
+
+    # upload to the target bucket
+    mdn_storage_client = storage.Client(args.destination_project)
+    mdn_bucket = mdn_storage_client.get_bucket(args.destination_bucket)
+    mdn_uri = f"{args.destination_path}/{target_file_name}.json"
+
+    logging.info("Uploading: %s to %s" % (local_path, mdn_uri))
+    blob = mdn_bucket.blob(mdn_uri)
+    blob.upload_from_filename(local_path)
```

Comment:
Why don't we write directly to the destination bucket?

Reply:
It's because we want the file to end up in a bucket within a different GCP project. I don't believe we can export the table directly from one project to a bucket inside another project.

Reply:
At least I was not able to find a nice way to do this. Please let me know if you know a way to handle this better.

Reply:
I think you can extract directly to a bucket in another project. Compare the GLAM DAG where we export to a bucket in `moz-fx-data-glam-prod-fca7`: https://github.com/mozilla/telemetry-airflow/blob/f5977b176aa3ca903efd83c751a6431b0554a96f/dags/glam_subdags/extract.py#L144

Reply:
ok turns out you can write to different projects, based on the docs it was not obvious to me and Google wasn't giving me any useful results. Made the change and removed the unnecessary middle steps.